### ISSUE TITLE
jestimator dependency needs updating

### ISSUE THREAD

**Issue by yhtang** [google-research/t5x#1294]
T5X currently depends on the [default version of jestimator](https://github.com/google-research/t5x/blob/1f0c4996dd2157298c1b6ab1176665af090af7ad/setup.py#L57) on PyPI, which is dated [Dec 5, 2022](https://pypi.org/project/jestimator/0.3.3/). This is no longer compatible with the latest JAX due to `jax.experimental.pjit.PartitionSpec` now becoming `jax.sharding.PartitionSpec`, leading to the error
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/opt/t5x/t5x/__init__.py", line 17, in <module>
    import t5x.adafactor
  File "/opt/t5x/t5x/adafactor.py", line 64, in <module>
    from t5x import utils
  File "/opt/t5x/t5x/utils.py", line 45, in <module>
    from t5x import checkpoints
  File "/opt/t5x/t5x/checkpoints.py", line 55, in <module>
    from t5x import optimizers
  File "/opt/t5x/t5x/optimizers.py", line 38, in <module>
    from jestimator import amos
  File "/usr/local/lib/python3.10/dist-packages/jestimator/__init__.py", line 20, in <module>
    from . import amos_helper
  File "/usr/local/lib/python3.10/dist-packages/jestimator/amos_helper.py", line 24, in <module>
    from jax.experimental.pjit import PartitionSpec
ImportError: cannot import name 'PartitionSpec' from 'jax.experimental.pjit' (/opt/jax-source/jax/experimental/pjit.py)
```

@yashk2810 @nouiz for viz

**Comment by yhtang** [comment]
Filed [issue](https://github.com/google-research/jestimator/issues/30) in jestimator repo to request a new PyPI release.

**Comment by adarob** [comment]
Fixed by #1299
